src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,37 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer producer = type.equals(Bulldozer.class) ? new BulldozerProducer()
+                : type.equals(Excavator.class) ? new ExcavatorProducer()
+                : type.equals(Truck.class) ? new TruckProducer()
+                : null;
```

Comment:
all the producers should be created once, not every time when this method is being called.
> the method getAll(Class type) produces the list of machines based on the input param.
> getAll(Class type)- based on the input class type, choose the right MachineProducer implementation and call its get() method.